### ISSUE TITLE
stickybits.cleanup() not working correctly?

### ISSUE THREAD

**Issue by hacknug** [dollarshaveclub/stickybits#188]
I'm using stikybits on a new website and I need to make sure it only applies the `position` and `top` properties when on a given breakpoint.

I've seen there's an example to the cleanup method (https://codepen.io/yowainwright/pen/gRgdep), the problem with this is that you're using a class selector that has no elements on the document that match (`.child-1`). The selector should be `.child` instead. First issue: solved.

Now, what I would expect is to be able to cleanup stickybits no matter what its config is. At least that what it seems you should be able to do when looking at this: https://github.com/dollarshaveclub/stickybits#stickybits-cleanup

The problem here is that stickybits doesn't clean anything unless you're using `useStickyClasses: true` (which is on the CodePen example I linked to earlier). This happens be because if you don't add this property, stickybits won't save the instances.

I think the conditional here should be changed to also save those instances to stickybits so we can clean them up and avoid having scroll listeners when we don't need to add/remove any CSS classes.

https://github.com/dollarshaveclub/stickybits/blob/f33c18c776187bd94ec315a21a45ba9243c3b1ee/src/stickybits.js#L85

----

Also it seems you have some redundant code in this two lines: https://github.com/dollarshaveclub/stickybits/blob/f33c18c776187bd94ec315a21a45ba9243c3b1ee/src/stickybits.js#L74 https://github.com/dollarshaveclub/stickybits/blob/f33c18c776187bd94ec315a21a45ba9243c3b1ee/src/stickybits.js#L112

**Comment by yowainwright** [comment]
@hacknug would you like to try and resolve the issue you're describing?

**Comment by hacknug** [comment]
Sure! Will get back to you when I have something 😉

**Comment by hacknug** [comment]
@yowainwright done! I opened a PR in #191

**Comment by yowainwright** [comment]
@hacknug this is actually not an issue I don't think. 🎉

Stickybits will be done (no more code will be run 😃 ) after it has set `position: sticky` in browsers that support it. To your valid point, if you need to remove the position sticky class—use the `useStickyClasses` prop.

**Comment by hacknug** [comment]
I don't see why it seems like a good idea to force anyone to use `useStickyClasses` just so they can use the `cleanup()`method. I strongly believe instances should get saved regardless of the value of `useStickyClasses`.

Using this only forces Stickybits to execute unneeded and unrelated code to what someone should be trying to achieve affecting the site's overall performance (not a huge impact, just adding to what in some cases could end up turning into a mess).

I'd be willing to take another look at the issue and try to fix it in a new PR if you're open to this. Let me know.

**Comment by denkristoffer** [comment]
Seeing the same behaviour as @hacknug. I agree that I would expect `stickybits.cleanup()` to remove anything related to stickybits, `useStickyClasses` or not. I don't want to add a scroll event listener just for this, so I've resorted to just removing all inline styles from the element for now instead of calling `cleanup()`.

**Comment by yowainwright** [comment]
@denkristoffer thanks for your thoughts. You do not need an event listener and leaving `position: sticky` there allows CSS to do what used to have to done with JS. 

This [link](https://developer.mozilla.org/en-US/docs/Web/CSS/position#sticky) on `position: sticky` might clarify something.

**Comment by denkristoffer** [comment]
@yowainwright Sorry, I probably should've explained my use case: I want to use `position: fixed` for all users, whether or not `position: sticky` is supported, until a media query matches (a specific viewport width) 🙂 After it matches I want to use `position: sticky` when possible. Basically, `position: fixed` for mobile users, regardless of whether or not `position: sticky` is supported, and `position: sticky` for desktop users when available 👍

**Comment by yowainwright** [comment]
@denkristoffer: 

- see #268
- also, [this](https://github.com/dollarshaveclub/stickybits/blob/master/src/stickybits.js#L70)
  - there is not documentation because there are not acceptance tests yet (aka "soft release"). I'm working on it.

**Comment by denkristoffer** [comment]
@yowainwright Am I way off if I only want Stickybits to activate after a certain viewport width? Sorry, I don't mean to be moaning about specific behaviour that you might not want to support, it's just not my impression that stickybits actually supports this. Please feel free to correct me if I am wrong 😃

**Comment by yowainwright** [comment]
Stickybits doesn't deal with window resize (viewport width).

**Comment by hacknug** [comment]
@denkristoffer not sure if this is going to be helpful or not but here it is anyway.

I also was trying to enable stickybits when matching a given media query and this is what I used (besides a _fixed_ version of stickybits, the one on #191 that ended up breaking other things and got reverted).

If screen is wider than 1366px, stickybits is called and the header element becomes sticky. If not, I destroy the stickybits instance, allowing whatever CSS is getting applied to do its own thing. Pretty simple and performant.

If you feel like this could help you and have any questions about how it works I can explain it a little better. Unfortunately I'm pretty busy working on other sites that don't depend on this library and can't put in the work to go through all the new great stuff Jeff's been adding lately.

Maybe at some point in the future if no one else tackles it.

```js
// Breakpoints
var mqL = window.matchMedia('(min-width: 1366px)')

// Ingredient Single Sticky
var stickyHeader
var ingredientHeaderSticky = function () {
  if (mqL.matches) {
    stickyHeader = stickybits('#ingredient > header', {
      stickyBitStickyOffset: headerHeight,
      useStickyClasses: true,
    })
  } else {
    if (stickyHeader) {
      stickyHeader.cleanup()
    }
  }
}
mqL.addListener(ingredientHeaderSticky)
ingredientHeaderSticky()
```
